### PR TITLE
Notify user when cluster reset is required

### LINKED ISSUE #103: Make sliders work out of the box
The sliders will change the settings in the settings files but RD doesn't apply them. If you restart rancher you get an error that requires running a console command to fix. This should be an intuitive experience.

### PR DESCRIPTION
**Pull request by mook-as**
This adds notifications at the bottom of the k8s page when a cluster reset is required because they changed memory / CPU settings.  Fixes #103.

image

### CODE REVIEW

**Comment by mattfarina** [comment]
@mook-as we should get this merged. It might be better to have an apply button as a follow-up.

Can you rebase and I'll review

**Comment by ericpromislow** [comment]
Two problems right off the bat:

1. I changed the desired memory to 5 GiB (and should we switch our UI everywhere to use GiB instead of GB?) but the change didn't go through.  Debugging showed that pressing the `Reset Kubernetes` button was invoking the slow reset.

2. So I modded the code to make pressing Shift-<Reset Kubernetes> force a slow change. The command-line now has `--memory=5g` as an arg to `minikube start` but I'm still getting the notification that the cluster needs to be reset, even though the memory pointer is still at 5. The shift-thing isn't good. I think we need to compare the current settings with the applied settings for memory and CPU, and do a slow reset if at least one of them differs.

Also, I think we can improve the notification - it's relatively hard to follow with all the `to`s in it, as in `The cluster must reset for changes to memory from 4 to 5 to take effect`.  Not exactly sure, maybe `The cluster needs to be reset to implement memory change from 4 to 5 GiB`.

**Comment by mook-as** [comment]
Rebased to resolve conflicts, but haven't resolved https://github.com/rancher-sandbox/rd/pull/132#issuecomment-799870108 yet — no need to review at this point.

**Comment by mook-as** [comment]
- Fixed issue where we were _not_ using a factor of 1024, hence always finding a cluster reset is required.
- Made slow reset mandatory when CPU / memory changes.
- Adjusted the wording on the notification; still not great though.

**Comment by mattfarina** [comment]
@mook-as can you please rebase?
